tools/x-repo-types/xrt.go

```diff
@@ -0,0 +1,261 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/types"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/tools/go/packages"
+
+	_ "embed"
+)
+
+//go:embed xrt_tmpl.go.tmpl
+var differTmpl string
+
+func main() {
+	var xPkg, xBranch, xType, yPkg, yBranch, yType string
+
+	rootCmd := &cobra.Command{
+		Use:   "xrt",
+		Short: "Compare types across repos",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := runApp(xPkg, xBranch, xType, yPkg, yBranch, yType); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	rootCmd.Flags().StringVar(&xPkg, "x-package", "", "Go repo and package for type x")
+	rootCmd.Flags().StringVar(&xBranch, "x-branch", "", "repository brahnch for type x")
+	rootCmd.Flags().StringVar(&xType, "x-type", "", "Exported type in the package for type x")
+	rootCmd.Flags().StringVar(&yPkg, "y-package", "", "Go repo and package for type for type y")
+	rootCmd.Flags().StringVar(&yBranch, "y-branch", "", "repository branch for type y")
+	rootCmd.Flags().StringVar(&yType, "y-type", "", "Exported type in the package for type y")
+
+	if err := rootCmd.Execute(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func runApp(xPkg, xBranch, xType, yPkg, yBranch, yType string) error {
```

Comment:
Do you need to create a temp dir to do all of this? I think this might do weird things if you run it inside a directory that already has a go.mod file.

Reply:
Yes, it added the SDK to go-algorand's go.mod when I run the make targets

Reply:
I'm open to a different approach. My understanding of how `go.mod` works is incomplete, so I suspect there is a better way.

Reply:
I'm going to resolve the suggestion. It doesn't seem like a big deal unless you expect people to run this on their workstation during normal development. The tool can be updated if this becomes a problem.